File: reverse-linked-list/reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if (!head || !head.next) return head
    let first = head, second = first.next
    while (second) {
        const temp = second.next
        second.next = first
        first = second
        second = temp
    }
    head.next = null
    return first
};

File: reverse-linked-list/reverse-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return head
    }
    
    var prev, next *ListNode
    curr := head
    
    for curr.Next != nil {
        next = curr.Next
        curr.Next = prev
        prev = curr
        curr = next
    }

    curr.Next = prev
    return curr
}